pkgs/by-name/qs/qsp/package.nix

```diff
@@ -0,0 +1,32 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  pkg-config,
+  oniguruma,
+}:
+stdenv.mkDerivation {
+  name = "qsp";
+  version = "0-unstable-2024-11-27";
+  src = fetchFromGitHub {
+    owner = "QSPFoundation";
+    repo = "qsp";
+    rev = "f6ede7f8756e49604de056fcbdfe99fa4abd4812";
+    sha256 = "MoNam2IFnLpk02tKp+lkl4l+mBiaWNPhFc3/n4zUHcw=";
+  };
+  buildInputs = [ oniguruma ];
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DUSE_INSTALLED_ONIGURUMA=ON"
+  ];
+  meta = {
+    description = "QuestSoft game engine";
```

Comment:
Missing `homepage`, `platforms`, possibly `mainProgram`

Reply:
Added homepage and platforms. It doesn't have a main program as it's a library.